applications/asset_tracker_v2/src/cloud/lwm2m_integration/lwm2m_integration.c

```diff
@@ -377,7 +377,7 @@ int cloud_wrap_init(cloud_wrap_evt_handler_t event_handler)
 	LOG_DBG("LwM2M endpoint name: %s", endpoint_name);
 
 	if (IS_ENABLED(CONFIG_LWM2M_INTEGRATION_FLUSH_SESSION_CACHE)) {
-		int fd = socket(AF_INET, SOCK_DGRAM, IPPROTO_DTLS_1_2);
+		int fd = zsock_socket(AF_INET, SOCK_DGRAM, IPPROTO_DTLS_1_2);
```

Comment:
I had a discussion with @rlubos about this. I can see the benefit of using zsock API in libraries, so that all users of the library are not forced to enable the POSIX subsystem. But, is there are reason why we should update all samples and applications in the SDK as well?

Reply:
I think that would be enough for the applications. Maybe we should have a look at the libraries, because many of those use the POSIX API. And it's not just the socket API, for example I think the date_time library uses POSIX clock APIs. But that's probably something we don't want to do in upmerge PR.